File: ios/GutHealthPackage/Sources/GutHealthFeature/Models/WeeklyInsight.swift
import Foundation

public struct WeeklyInsight: Codable, Identifiable, Sendable {
    public var id: String?
    public var weekStart: String
    public var weekEnd: String
    public var gutScoreTrend: [Int]
    public var avgGutScore: Double
    public var topTriggers: [String]
    public var improvements: [String]
    public var aiReport: String
    public var generatedAt: Date

    public init(id: String? = nil, weekStart: String, weekEnd: String, gutScoreTrend: [Int] = [], avgGutScore: Double = 0, topTriggers: [String] = [], improvements: [String] = [], aiReport: String = "", generatedAt: Date = Date()) {
        self.id = id
        self.weekStart = weekStart
        self.weekEnd = weekEnd
        self.gutScoreTrend = gutScoreTrend
        self.avgGutScore = avgGutScore
        self.topTriggers = topTriggers
        self.improvements = improvements
        self.aiReport = aiReport
        self.generatedAt = generatedAt
    }
}

File: android/app/src/main/java/com/twintipsolutions/guthealth/data/models/WeeklyInsight.kt
package com.twintipsolutions.guthealth.data.models

import com.google.firebase.Timestamp

data class WeeklyInsight(
    val weekStart: String = "",
    val weekEnd: String = "",
    val gutScoreTrend: List<Int> = emptyList(),
    val avgGutScore: Double = 0.0,
    val topTriggers: List<String> = emptyList(),
    val improvements: List<String> = emptyList(),
    val aiReport: String = "",
    val generatedAt: Timestamp? = null
)
